File: Python/Medium/like_or_dislike.py
def like_or_dislike(lst):
    f="Nothing"
    if not(lst):
        return f
    else:
        for i in lst:
            if f!=i:
                f=i
            else:
                f="Nothing"
    return f

print(like_or_dislike(['Dislike']), 'Dislike')
print(like_or_dislike(['Like', 'Like']), 'Nothing')
print(like_or_dislike(['Dislike', 'Dislike']), 'Nothing')
print(like_or_dislike(['Like', 'Like', 'Like']), 'Like')
print(like_or_dislike(['Like', 'Dislike']), 'Dislike')
print(like_or_dislike(['Dislike', 'Like']), 'Like')
print(like_or_dislike(['Like', 'Dislike', 'Dislike']), 'Nothing')
print(like_or_dislike(['Dislike', 'Like', 'Dislike']), 'Dislike')
print(like_or_dislike([]), 'Nothing')

File: javascript/Easy/likeOrDislike.js
//https://www.codewars.com/kata/62ad72443809a4006998218a/train/javascript

const likeOrDislike = (buttons) => {
  return buttons.reduce((p, c) => (p === c ? Nothing : c), Nothing);
};

console.log(likeOrDislike([Dislike]), Dislike);
console.log(likeOrDislike([Like, Like]), Nothing);
console.log(likeOrDislike([Dislike, Like]), Like);
console.log(likeOrDislike([Like, Dislike, Dislike]), Nothing);

console.log(likeOrDislike([Dislike, Dislike]), Nothing);
console.log(likeOrDislike([Like, Like, Like]), Like);
console.log(likeOrDislike([Like, Dislike]), Dislike);
console.log(likeOrDislike([Dislike, Like, Dislike]), Dislike);
console.log(
  likeOrDislike([Like, Like, Dislike, Like, Like, Like, Like, Dislike]),
  Dislike
);
console.log(likeOrDislike([]), Nothing);
